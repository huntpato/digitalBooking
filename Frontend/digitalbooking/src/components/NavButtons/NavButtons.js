import React, {useState, useEffect, useContext} from "react";
import { Button } from "../Button/Button";
import styles from "./NavButtons.module.css";
import { useLocation } from 'react-router-dom';
import { NavContext } from "../Nav/NavContext";

export const NavButtons = (props) => {
  const { registerButtons, hide } = styles;
  const {signedbar, setSignedbar}= useContext(NavContext);
  const location = useLocation();
  const [login, setlogin] = useState();
  const [register, setregister] = useState();
 
  useEffect(() => {
    if(location.pathname === '/login'){
      setlogin(true);
      setregister(false);
    }else if(location.pathname === '/register'){
      setlogin(false);
      setregister(true);
    }else{
      setlogin(false);
      setregister(false);
    }
  }, [location, login, register, setlogin, setregister])

  
  return (
    <div className= {registerButtons}>
      <Button visibility= { register } text={"Crear cuenta"} link="/register" />
      <Button visibility= { login } className= {register && {hide}} text={"Iniciar sesion"} link="/login" />
    </div>
  );
};
