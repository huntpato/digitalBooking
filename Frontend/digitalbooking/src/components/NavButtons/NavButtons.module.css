
.hide{
  display: none !important;
  visibility:hidden;

}

.registerButtons {
  display: flex;
  flex-direction: row;
  margin-right: 40px;
}



  
@media (max-width: 768px) {
  .registerButtons {
    margin-right: 3%;
  }
}

@media (max-width: 480px) {
  .registerButtons {
    display: none;
  }
}
