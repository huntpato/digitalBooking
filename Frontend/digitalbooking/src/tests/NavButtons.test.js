import React from "react";
import { mount, shallow } from "enzyme";
import { NavButtons } from "../components/NavButtons/NavButtons";
import "@testing-library/jest-dom";
import { expect } from "@jest/globals";

jest.mock("react-router-dom", () => ({
  useLocation: jest.fn().mockReturnValue({
    pathname: "localhost:3000/login",
    search: "",
    hash: "",
    state: null,
    key: "5nvxpbdafa",
  }),
}));


describe("Pruebas en componente <NavButtonss />", () => {
  const wrapper = shallow(<NavButtons />);

  test("Comprobar que <NavButtonss /> renderiza correctamente.", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Verificar que el componente carga la clase de CSS correctamente", () => {
    const div = wrapper.find("div");
    expect(div.hasClass("registerButtons")).toBe(true);
  });

  test("Verificar que el componente carga el boton de register", () => {
    const div = wrapper.find("Button").at(0);
    expect(div.prop('text')).toBe('Crear cuenta');
  });

  test("Verificar que el componente carga el boton de login", () => {
    const div = wrapper.find("Button").at(1);
    expect(div.prop('text')).toBe('Iniciar sesion');
  });

});
