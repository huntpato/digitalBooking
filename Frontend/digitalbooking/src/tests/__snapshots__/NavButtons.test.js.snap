// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <NavButtonss /> Comprobar que <NavButtonss /> renderiza correctamente. 1`] = `
<div
  className="registerButtons"
>
  <Button
    link="/register"
    text="Crear cuenta"
  />
  <Button
    link="/login"
    text="Iniciar sesion"
  />
</div>
`;
